pkg/app/piped/executor/analysis/metrics_analyzer.go

```diff
@@ -160,6 +169,9 @@ func (a *metricsAnalyzer) analyzeWithPrevious(ctx context.Context) (bool, error)
 	}
 
 	prevMetadata, err := a.analysisResultStore.GetLatestAnalysisResult(ctx)
+	if errors.Is(err, analysisresultstore.ErrNotFound) {
+		return false, fmt.Errorf("seems like this is the first deployment: %w", errShouldEndWithSuccess)
```

Comment:
tbh, I don't really like this `errShouldEndWithSuccess` error even I get what you want to do here. Should we add one more return value to this function signature for this case, since this check is about to compare with the previous, compare with the previous in case of the first deployment should be treated as a special case.
I mean something like `func (a *metricsAnalyzer) analyzeWithPrevious(ctx context.Context) (bool, bool, error)`

Reply:
I know what you mean. Well, let me apply it 👍